spec/github_helper_spec.rb

```diff
@@ -217,31 +217,140 @@ def get_milestone(milestone_name:)
       allow(Octokit::Client).to receive(:new).and_return(client)
     end
 
-    it 'has the correct dates to code freeze without submission' do
-      comment = "Code freeze: October 22, 2022\nApp Store submission: November 15, 2022\nRelease: October 25, 2022\n"
-      options = { due_on: '2022-10-22T12:00:00Z', description: comment }
+    it 'computes the correct dates for standard period' do
+      due_date = '2022-12-02T08:00:00Z'.to_time.utc
+      options = {
+        due_on: '2022-12-02T12:00:00Z',
+        description: "Code freeze: December 02, 2022\nApp Store submission: December 06, 2022\nRelease: December 09, 2022\n"
+      }
 
       expect(client).to receive(:create_milestone).with(test_repo, test_milestone_number, options)
-      create_milestone(need_submission: false, milestone_duration: 24, days_code_freeze: 3)
+      create_milestone(due_date: due_date, days_until_submission: 4, days_until_release: 7)
     end
 
-    it 'has the correct dates to code freeze with submission' do
-      comment = "Code freeze: October 22, 2022\nApp Store submission: October 22, 2022\nRelease: October 25, 2022\n"
-      options = { due_on: '2022-10-22T12:00:00Z', description: comment }
+    it 'computes the correct dates when submission and release dates are in the same day' do
+      due_date = '2022-12-02T08:00:00Z'.to_time.utc
+      options = {
+        due_on: '2022-12-02T12:00:00Z',
+        description: "Code freeze: December 02, 2022\nApp Store submission: December 03, 2022\nRelease: December 03, 2022\n"
+      }
 
       expect(client).to receive(:create_milestone).with(test_repo, test_milestone_number, options)
-      create_milestone(need_submission: true, milestone_duration: 19, days_code_freeze: 3)
+      create_milestone(due_date: due_date, days_until_submission: 1, days_until_release: 1)
     end
 
-    def create_milestone(need_submission:, milestone_duration:, days_code_freeze:)
+    it 'computes the correct dates when the due date is on the verge of a DST day change' do
+      # Europe DST starts on the last Sunday of March, and ends on the last Sunday of October
+      Time.use_zone('Europe/London') do
+        due_date = Time.zone.parse('2022-03-27 23:00:00Z')
+        options = {
+          due_on: '2022-03-28T12:00:00Z',
```

Comment:
Nice edge case of having `2022-03-27 23:00:00Z` being exactly `2022-03-28 00:00:00 +0100` because that day is the exact day of the DST change, when London has switched to `+0100`… while for your next test case on line 259, the `2022-03-26 23:00:00Z` date still corresponds to `2022-03-26 23:00:00 +0000` because this is just before the DST change when London were still on `+0000`…

Took me a while to figure that out though and to triple-check that this difference between the `due_date` (2022-03-27` and the `due_on` (`2022-03-28`) was the correct and expected behavior here, and not an incorrect value due to our date math being buggy or whatnot. Phew!

Could maybe benefit from some small code comment (here and on line 259) to give more context about those tricky details?

Reply:
For sure! 😄 

Besides the comments, I also added another test to cover the scenario of the `due_date` is not close enough to the end of the day to exist a date change (Line 259) and also add a new case, to test what happens if you have a date close to the day change, but not in DST (Line 278).